src/main/java/org/jenkinsci/plugins/discardbuild/DiscardBuildPublisher.java

```diff
@@ -216,6 +218,22 @@ public boolean add(Run<?, ?> run) {
         return newList.getNewList();
     }
 
+    private ArrayList<Run<?, ?>> HoldMaxBuilds(ArrayList<Run<?, ?>> listIn, BuildListener listener, int maxCount) {
+        ArrayList<Run<?, ?>> listUpdt = new ArrayList<Run<?, ?>>();
+        listUpdt = listIn;
+        int listCnt = listUpdt.size();
+        if (listCnt < maxCount||listCnt == maxCount){ // clear discard list if beneath minimum build quantity
+            listener.getLogger().println("Too few builds present to remove any, clearing discard list.");
+            listUpdt.clear();}
+        else if (listCnt > maxCount) {
+            for (int i = 0; i < maxCount; i++) {
+                listener.getLogger().println("Removing build from discard list to maintain max quantity:");
+                listener.getLogger().println(listUpdt.get(0));
```

Comment:
This might produce a very long list in the console. I suggest printing the range outside the loop instead.